bc_gym_planning_env/envs/synth_turn_env.py

```diff
@@ -412,8 +412,12 @@ def _extract_egocentric_observation(self, rich_observation):
         obs = np.expand_dims(ego_costmap.get_data(), -1)
         normalized_goal = ego_path[-1, :2] / ego_costmap.world_size()
         normalized_goal = normalized_goal / np.linalg.norm(normalized_goal)
+
+        robot_state = rich_observation.robot_state.to_numpy_array()
+        goal_n_state = np.hstack([normalized_goal, robot_state[3:]])
```

Comment:
why cut state to 3:?
If this is because it improves performance of continuous PPO, then it would be nicer to do this kind of filtering somewhere else.

Reply:
haha, actually I never tested with the whole thing. The first 3 items are x, y, angle, since this is in egocentric frame so I left them out when I first wrote this. I am running a job to test if this matters. We will know tomorrow :)

Reply:
turns out it does matter......I've refactored this as requested. please review

Reply:
now `goal_n_state` will have different dimension, depending on whether we are using tricycle or diff drive. This will cause problems with number 5 in `('goal', spaces.Box(low=-1., high=1., shape=(5, 1), dtype=np.float64))`, right?

Reply:
good call. How about we add a dummy 0.0 to the diff drive function?